src/Raven.Server/Documents/Handlers/TransactionsRecordingHandler.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Operations.TransactionsRecording;
+using Raven.Client.Documents.Session;
+using Raven.Server.Json;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers
+{
+    public class TransactionsRecordingHandler : DatabaseRequestHandler
+    {
+        [RavenAction("/databases/*/transactions/replay", "POST", AuthorizationStatus.ValidUser)]
+        public async Task ReplayRecording()
+        {
+            var operationId = GetLongQueryString("operationId");
+            using (var operationCancelToken = CreateOperationToken())
+            {
+                var replayStream = RequestBodyStream();
+
+                var result = await Database.Operations.AddOperation(
```

Comment:
Do we want this endpoint to be synchronous? Or return 'OperationId' ? It looks like it could be async with OperationId?

Reply:
You can't make this aysnc, the file is still being streamed upward to the server

Reply:
True